src/harness/testcases/WINNF_FT_S_SIQ_testcase.py

```diff
@@ -841,6 +842,237 @@ def test_WINNF_FT_S_SIQ_11(self):
+
+        # Check if the ruleApplied field is set to FCC_PART_96
+        for channel in response['availableChannel']:
+          self.assertEqual(channel['ruleApplied'], 'FCC_PART_96')
+
+        # Check that the GAA channel frequency ranges do not overlap with FR1cbsd ranges
+        for channel in response['availableChannel']:
+          if channel['channelType'] == 'GAA':
+            for fr1_range in config['fr1Cbsd'][index]:
+              # Check if the channel range is lower than the low freq of FR1cbsd range or if it is higher 
+              # than the high freq. If not then it is considered to be overlapping
+              self.assertTrue(((channel['frequencyRange']['highFrequency'] <= fr1_range['frequency']['lowFrequency']) or 
+                               (channel['frequencyRange']['lowFrequency'] >= fr1_range['frequency']['highFrequency'])), 
+                                "GAA channel overlaps with FR1cbsd")
+
+        # Check the PAL channels frequency ranges overlap with FR2cbsd
+        for channel in response['availableChannel']:
+          if channel['channelType'] == 'PAL':
+            isFrequencyIncludedInFr2Cbsd = False
+            for fr2_range in config['fr2Cbsd'][index]:
+              if ((channel['frequencyRange']['lowFrequency'] ==
+                             fr2_range['frequency']['lowFrequency']) and
+                  (channel['frequencyRange']['highFrequency'] ==
+                             fr2_range['frequency']['highFrequency'])):
+                isFrequencyIncludedInFr2Cbsd = True
+            self.assertTrue(isFrequencyIncludedInFr2Cbsd, "PAL channel is not included in Fr2Cbsd list")       
+  
+            # Check the PAL channels frequency ranges does not include any frequency range from 3650-3700MHz.
+            self.assertLess(channel['frequencyRange']['lowFrequency'], 3650000000) 
+            self.assertLessEqual(channel['frequencyRange']['highFrequency'], 3650000000) 
+
+        # Check the available channels are within the range requested by that CBSD
+        for channel in response['availableChannel']:
+          isFrequencyIncludedInRange = False
+          for inquired_spectrum in spectrum_inquiry_request['spectrumInquiryRequest'][index]['inquiredSpectrum']:
+            if ((channel['frequencyRange']['lowFrequency'] >= inquired_spectrum['lowFrequency']) and 
+                (channel['frequencyRange']['lowFrequency'] < inquired_spectrum['highFrequency']) and
+                (channel['frequencyRange']['highFrequency'] > inquired_spectrum['lowFrequency']) and
+                (channel['frequencyRange']['highFrequency'] <= inquired_spectrum['highFrequency'])):
+              isFrequencyIncludedInRange = True
+          self.assertTrue(isFrequencyIncludedInRange, "Available channel is not in requested frequency range")
+
+      else:
+        # Check 'availableChannel' is not present if the response code is not SUCCESS
+        self.assertTrue('availableChannel' not in response)
```

Comment:
`self.assertFalse('availableChannel' in response)` to make debugging easier.

Reply:
Agreed. Modified.